Index: nova/nova/rootwrap/volume.py
===================================================================
--- nova.orig/nova/rootwrap/volume.py	2012-07-11 08:40:49.000000000 +0200
+++ nova/nova/rootwrap/volume.py	2012-07-11 08:41:49.000000000 +0200
@@ -41,5 +41,5 @@
 
     # nova/volume/driver.py: 'iscsiadm', '-m', 'discovery', '-t',...
     # nova/volume/driver.py: 'iscsiadm', '-m', 'node', '-T', ...
-    filters.CommandFilter("/sbin/iscsiadm", "root"),
+    filters.CommandFilter("/usr/bin/iscsiadm", "root"),
     ]
