Description: Fixe path to nova.conf in nova-manage
Author: unkonwn
Forwarded: no

--- a/bin/nova-manage
+++ b/bin/nova-manage
@@ -1659,7 +1659,10 @@
 
 def main():
     """Parse options and call the appropriate class/method."""
-    flagfile = utils.default_flagfile()
+    cfg = "/etc/nova/nova.conf"
+    if not (os.path.exists(cfg) and os.access(cfg, os.R_OK)):
+        cfg = "nova.conf"
+    flagfile = utils.default_flagfile(cfg)
 
     if flagfile and not os.access(flagfile, os.R_OK):
         st = os.stat(flagfile)
