Description: Fixes command line for ajaxterm
Forwarded: not-needed
Author: Soren Hansen <soren@ubuntu.com>
Last-Update: 2011-04-15
--- a/nova/virt/libvirt/utils.py
+++ b/nova/virt/libvirt/utils.py
@@ -234,8 +234,9 @@
     :param token: Token to require for authentication
     :param port: Port to run on
     """
-    cmd = ['%s/tools/ajaxterm/ajaxterm.py' % utils.novadir(),
-           '--command', cmd, '-t', token, '-p', port]
+    cmd = ['ajaxterm', '--command', ajaxterm_cmd, '-t', token,
+           '-p', port, '-T', '300']
+
     execute(*cmd)
 
 
