Description: Fixes the path to the xenhost.conf file
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
--- a/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost
+++ b/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost
@@ -40,7 +40,7 @@
 pluginlib.configure_logging("xenhost")
 
 host_data_pattern = re.compile(r"\s*(\S+) \([^\)]+\) *: ?(.*)")
-config_file_path = "/usr/etc/xenhost.conf"
+config_file_path = "/etc/xcp/xenhost.conf"
 
 
 def jsonify(fnc):
