Description: Fixes the path to the xenhost.conf file
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
Index: nova/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost
===================================================================
--- nova.orig/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost	2015-02-08 15:44:58.000000000 +0100
+++ nova/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost	2015-02-08 15:44:59.000000000 +0100
@@ -46,7 +46,7 @@
 
 
 host_data_pattern = re.compile(r"\s*(\S+) \([^\)]+\) *: ?(.*)")
-config_file_path = "/usr/etc/xenhost.conf"
+config_file_path = "/etc/xcp/xenhost.conf"
 DEFAULT_TRIES = 23
 DEFAULT_SLEEP = 10
 
