Description: Fixes the path to the xenhost.conf file
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
Index: nova/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost
===================================================================
--- nova.orig/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost	2013-09-12 21:33:58.000000000 +0800
+++ nova/plugins/xenserver/xenapi/etc/xapi.d/plugins/xenhost	2013-09-12 21:33:59.000000000 +0800
@@ -41,7 +41,7 @@
 
 
 host_data_pattern = re.compile(r"\s*(\S+) \([^\)]+\) *: ?(.*)")
-config_file_path = "/usr/etc/xenhost.conf"
+config_file_path = "/etc/xcp/xenhost.conf"
 DEFAULT_TRIES = 23
 DEFAULT_SLEEP = 10
 
